src/js/draad-maps.js

```diff
@@ -353,17 +385,30 @@ class Draad_Map {
 	 * @param {string} state The state of the feature.
 	 */
 	dataSetState = (feature, state) => {
-		const style = feature._style;
+		if (typeof feature.setStyle !== "function") return;
+		
+		let style = { ...feature._style };
+		const originalOpacity = !isNaN( feature._style.fillOpacity ) ? parseFloat(feature._style.fillOpacity) : 0.3;
+
 		switch (state) {
 			case "active":
+				feature.isActive = true; // Flag to persist active state
+				style.fillOpacity = Math.min(originalOpacity + 0.2, 0.8); // More visible (e.g., 0.5 -> 1.0)
+				break;
 			case "hover":
+				if (feature.isActive) return; // Don't override active
+				style.fillOpacity = Math.min(originalOpacity + 0.2, 0.8); // Moderate increase (e.g., 0.5 -> 0.75)
+				break;
 			case "focus":
-				style.fillOpacity = 0.15;
+				if (feature.isActive) return; // Don't override active
+				style.fillOpacity = Math.min(originalOpacity + 0.2, 0.8); // Same as hover for consistency
 				break;
```

Comment:
De `hover` en `focus` zijn nu hetzelfde, wellicht samenvoegen onder dezelfde case?


```suggestion
case "hover":
case "focus":
				if (feature.isActive) return; // Don't override active
				style.fillOpacity = Math.min(originalOpacity + 0.2, 0.8); // Moderate increase (e.g., 0.5 -> 0.75)
break;
```